lms/djangoapps/grades/rest_api/v1/tests/test_views.py

```diff
@@ -257,3 +260,843 @@ def test_staff_can_get_all_grades(self):
+                                                               'detail': 'Lab 9 Unreleased - 0% (?/?)',
+                                                               'label': 'Lab 09',
+                                                               'percent': 0.0},
+                                                              {'category': 'Lab',
+                                                               'detail': 'Lab 10 Unreleased - 0% (?/?)',
+                                                               'label': 'Lab 10',
+                                                               'percent': 0.0},
+                                                              {'category': 'Lab',
+                                                               'detail': 'Lab 11 Unreleased - 0% (?/?)',
+                                                               'label': 'Lab 11',
+                                                               'mark': {'detail': 'The lowest 2 Lab scores are dropped.'},  # nopep8 pylint: disable=C0301
+                                                               'percent': 0.0},
+                                                              {'category': 'Lab',
+                                                               'detail': 'Lab 12 Unreleased - 0% (?/?)',
+                                                               'label': 'Lab 12',
+                                                               'mark': {'detail': 'The lowest 2 Lab scores are dropped.'},  # nopep8 pylint: disable=C0301
+                                                               'percent': 0.0},
+                                                              {'category': 'Lab',
+                                                               'detail': 'Lab Average = 0%',
+                                                               'label': 'Lab Avg',
+                                                               'percent': 0.0,
+                                                               'prominent': True},
+                                                              {'category': 'Midterm Exam',
+                                                               'detail': 'Midterm Exam = 0%',
+                                                               'label': 'Midterm',
+                                                               'percent': 0.0,
+                                                               'prominent': True},
+                                                              {'category': 'Final Exam',
+                                                               'detail': 'Final Exam = 0%',
+                                                               'label': 'Final',
+                                                               'percent': 0.0,
+                                                               'prominent': True}],
+                                          'username': 'student'}])])
```

Comment:
It's hard to read this. Would it make sense to extract this data to a class attribute and reuse it?

We could think about an easy generator, like:
```python
[{'category': 'Homework', 'detail': f'Homework {i} Unreleased - 0% (?/?)', 'label': f'HW {i:02d}', 'percent': .0} for i in range(12)] + [{'category': 'Homework', 'detail': 'Homework Average = 0%', 'label': 'HW Avg', 'percent': 0.0, 'prominent': True}] + ...
```